### DISCUSSION TITLE
How do I use the Xamarin.Forms source code to work around a known bug?

### DISCUSSION THREAD

**Discussion by tpitman** [xamarin/Xamarin.Forms#14519] [Q&A]
There is a bug that already has identified what is causing the issue. It has to do with the WebView and a null crash.

It doesn't look like it is going to be released any time soon. I was wondering how I would go about using the XF source code so that I could try and fix it myself for my use.

I was looking at how to build and that seems to focus on how to contribute to the project. As much as I would love to do that I am not in a position to help.

I really just need this one bug fixed and don't know how to go about getting just that fix into my project.

Can someone help me figure out how?

The specific bug is this one:

https://github.com/xamarin/Xamarin.Forms/issues/13326

**Comment by pictos** [comment]
Hey, @tpitman you can do a Custom Renderer that extends the webView control and place your fix/workaround there. If that's enough, like you need to change something that can't be overridden, you can copy all the Xamarin.Forms webview implementation code and fix there, it makes sense?
That will be much easier than maintaining your own Xamarin.Forms version

**Reply by tpitman** [reply]
Thank you for the reply. I will see if I can figure out how to do that. I will have to look and see where in the code it is happening.

**Reply by tpitman** [reply]
So the problem code is in Xamarin.Forms/Xamarin.Forms.Core/WebView.cs. Would I just copy that single file into my project and build it?

If not what all would I have to copy into my project and what don't I have to copy?

**Reply by pictos** [reply]
Hey @tpitman sorry for the late reply.

Actually no, you need to copy the platform-specific code for the renderer, for android will be [this one](https://github.com/xamarin/xamarin.forms/blob/5.0.0/Xamarin.Forms.Platform.Android/Renderers/WebViewRenderer.cs) and iOS [this one](https://github.com/xamarin/xamarin.forms/blob/5.0.0/Xamarin.Forms.Platform.iOS/Renderers/WkWebViewRenderer.cs). 
Probably some methods will be marked as errors because they are internal, so you need to find them as well and implement them on your own. When you have all ready you will be able to debug it and fix any issue that you may have
